cabal-install/src/Distribution/Client/Dependency.hs

```diff
@@ -713,7 +735,7 @@ standardInstallPolicy
   -> [PackageSpecifier UnresolvedSourcePackage]
   -> DepResolverParams
 standardInstallPolicy installedPkgIndex sourcePkgDb pkgSpecifiers =
-  addDefaultSetupDependencies mkDefaultSetupDeps $
+  addDefaultSetupDependencies setImplicitSetupInfo mkDefaultSetupDeps $
```

Comment:
Here, are we doing the right thing? We only add implicit dependencies to implicit setup stanzas, but do we also want to add hooks-exe as an implicit dependency to Hooks buildtypes here?